ldap_auth_provider.py

```diff
@@ -430,6 +443,8 @@ async def _fetch_root_domain(self) -> str:
             logger.warning("Unable to get root domain due to failed LDAP bind")
             return self.ldap_root_domain
 
+        assert isinstance(conn, ldap3.Connection)
```

Comment:
Are you asserting that this is not `None` here? We usually do that with:

```suggestion
        assert conn is not None
```